src/server/pps/server/master.go

```diff
@@ -351,36 +351,37 @@ func (a *apiServer) upsertWorkersForPipeline(pipelineInfo *pps.PipelineInfo) err
 	return nil
 }
 
-func (a *apiServer) deleteWorkersForPipeline(pipelineInfo *pps.PipelineInfo) error {
-	cancel, ok := a.monitorCancels[pipelineInfo.Pipeline.Name]
+func (a *apiServer) deleteWorkersForPipeline(pipelineName string) error {
+	cancel, ok := a.monitorCancels[pipelineName]
 	if ok {
 		cancel()
-		delete(a.monitorCancels, pipelineInfo.Pipeline.Name)
+		delete(a.monitorCancels, pipelineName)
 	}
-	rcName := ppsutil.PipelineRcName(pipelineInfo.Pipeline.Name, pipelineInfo.Version)
-	if err := a.kubeClient.CoreV1().Services(a.namespace).Delete(
-		rcName, &metav1.DeleteOptions{},
-	); err != nil {
-		if !isNotFoundErr(err) {
-			return err
-		}
+	selector := fmt.Sprintf("pipelineName=%s", pipelineName)
```

Comment:
So using this label selector matches all the k8s objects across pipeline versions and pipelines that are services? That's a lot simpler

Reply:
Yup, much more foolproof and requires less information too.